### DISCUSSION TITLE
5주차 발표 자료_Peri-LN: Revisiting Layer Normalization in the Transformer Architecture_네이버클라우드_김정훈

### DISCUSSION THREAD

**Discussion by kimc-sqzb** [efficient-ai-study/efficient-ai-study#26] [2025 Seminar]
[5주차 김정훈님 발표자료_Peri-LN_ Revisiting Layer Normalization in the Transformer Architecture.pdf](https://github.com/user-attachments/files/20708939/5._Peri-LN_.Revisiting.Layer.Normalization.in.the.Transformer.Architecture.pdf)

**Comment by penpaperkeycode** [comment]
아래와 같이 질문에 대한 답변을 드립니다! 

>Peri-LN에서 output을 normalization하면 massive activaiton이 줄어든다고 말씀하셨는데 그렇다면 실제로 gemma2나 gemma3 또는 학습한 모델의 activation의 크기가 llama에 비해 많이 줄어들었는지 궁금합니다.

1. Gemma2는 massive activation 페이퍼에서 말하는 현상은 발견되지 않습니다.   massive activation paper에서도 같은 현상을 관찰했을것으로 예상되는데요. 그래서인지 gemma familiy는 언급을 안하고 있는것으로 알고 있습니다. 
2. 다만 Gemma3는 massive activation과 비슷한 현상이 발생하고 있습니다. 정확히는 module output은 normalize되기 때문에 여기서는 발생하지 않으나, RMSNorm의 gamma scale이 비정상적으로 큰 값을 갖고 있습니다. RMSNorm gamma scale에 정확히 어떤 일이 일어났는지는 모르겠지만 이 학습 당시에 weight decay가 아예 안들어갔거나 굉장히 작은 값으로 설정되었을것으로 예측해볼 수 있을것 같습니다.
3. 논문의 Proposition에서 얻은 내용을 바탕으로 논의해보면, massive activations가 training stability에 영향을 주려면, Attention & MLP module output을 Normalize하고있지 않을 경우여야합니다. 이 경우, gradient norm이 hidden state의 크기에 비례하여 갑자기 큰 값이 업데이트 될 수 있기 때문입니다. 그러나 Gemma2&3의 경우는 gradient norm이 output normalization에 의해 bounded 라서 큰 값의 hidden state가 발생하더라도 각 업데이트 될 모듈에서 gradient norm이 갑자기 커지는 현상은 보기 어려울 것이라고 예상해볼 수 있습니다. 

> weight decay를 통해 massive activation을 어느 정도 방지할 수 있다는 내용을 말씀해주셨는데 해당 부분이 어떻게 방지가 되는지 궁금합니다.

이 부분은 [Peri-LN paper](https://arxiv.org/pdf/2502.02732)의 section 6.1 을 참고해주시면 좋을것 같습니다. 구체적인 메커니즘이 궁금하신것이라면, 요건 제가 현재 후속 연구로 진행하는 내용과 관련이 있을것 같은데요. Weight decay가 massive activations를 줄여주는 메커니즘은 아래와 같이 설명해볼 수 있을것 같습니다. 
<img width="995" alt="image" src="https://github.com/user-attachments/assets/f571169f-4530-4e4e-9892-8f137da797a1" />

혹시 더 궁금하신 부분이 있으시면 언제든 편히 알려주세요!
